flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -723,14 +724,22 @@ else if (!props.forceCheckpoint()) {
 	 * Receives a {@link DeclineCheckpoint} message for a pending checkpoint.
 	 *
 	 * @param message Checkpoint decline from the task manager
+	 * @param taskManagerLocation The location of the decline checkpoint message's sender
 	 */
-	public void receiveDeclineMessage(DeclineCheckpoint message) {
+	public void receiveDeclineMessage(DeclineCheckpoint message, TaskManagerLocation taskManagerLocation) {
 		if (shutdown || message == null) {
 			return;
 		}
+
+		if (taskManagerLocation == null) {
+			throw new IllegalArgumentException("Received DeclineCheckpoint message " + message.getCheckpointId() +
+				" by task " + message.getTaskExecutionId() + " of job " +
+				message.getJob() + " while can not find the task's location information ");
+		}
```

Comment:
We must not fail if some information needed for logging purposes only is not available.